cEP-0016.md

```diff
@@ -0,0 +1,382 @@
+             yield Result(self, 'The CI engine for this project does not allow skipping CI builds.')
+
+        test_files = # get all test files
+        coalafile_files = # get all coalafile files
+
+        modified_files = # get filename of files modified by this commit
+
+        for filename in modified_files:
+            if filename in coafile_files or filename in test_files:
+                yield Result(self, 'Remove [ci skip] from commit message.')
+```
+
+A prototype of `AnalyzeCommitBear` :
+
+```python
+    import re
+
+    from bears.vcs.CommitBear import _CommitBear
+    from coala_utils.ContextManagers import change_directory
+    from coalib.misc.Shell import run_shell_command
+    from unidiff import PatchSet
+
+    class AnalyzeCommitBear(CommitBear):
+        BEAR_DEPS = {'_CommitBear'}
+        LANGUAGES = {'Git'}
+        CAN_DETECT = {'Modifications in different files'}
+
+        def run(self, **kwargs):
+            """
+            Check the current commit message at HEAD.
+
+            Return lists of modified files with correspoding
+            number of additions, deletions and modifications in each file.
+
+            """
+            commit_hash = # get commit hash of HEAD commit
+
+            yield from self.get_files_modified_by_commit(
+                commit_hash)
+
+        def get_files_modified_by_commit(self):
+            command = 'git show --pretty="" --name-only' + str(commit_hash)
+            modified_files = run_shell_command(command)
+
+            encoding = diff.headers.get_charsets()[0]
+
+            for filename in modified_files:
+                diff = filename + '.diff'
+                patch = PatchSet.from_filename(diff, encoding=encoding)
+
+            # Return lists of modified, added & deleted files
+```
+
+A prototype of `MergeCommitBear` :
+
+```python
+    import re
+
+    from bears.vcs.CommitBear import _CommitBear
+    from coala_utils.ContextManagers import change_directory
+    from coalib.misc.Shell import run_shell_command
+    from unidiff import PatchSet
+
+    class MergeBear(CommitBear):
+        BEAR_DEPS = {'_CommitBear'}
+        LANGUAGES = {'Git'}
+        CAN_DETECT = {'Modifications in different files'}
+
+        def run(self, **kwargs):
+            """
+            Returns a list of heads of merge commit.
+            """
+            commit_hash = # get commit hash of HEAD commit
+
+            yield from self.get_files_modified_by_commit(
+                commit_hash)
+
+        def get_heads_of_merge_commit(self, commit_hash):
+            # Here we get all the heads of this merge commit
+```
+
+A prototype of `RevertCommitBear` :
+
+```python
+    import re
+
+    from bears.vcs.CommitBear import _CommitBear
+    from coala_utils.ContextManagers import change_directory
+    from coalib.misc.Shell import run_shell_command
+    from unidiff import PatchSet
+
+    class RevertCommitBear(CommitBear):
```

Comment:
`CommitBear` -> `_CommitBear` for all these bears

Reply:
Also if those bears are to be meta-bears, (and not callable directly), they should start with an _

Reply:
no, `_` is for which you did, and isnt a 'thing' in coala terminology, but might be called an abstract bear.

this project is supposed to be doing 'meta' bears, like cEP-0025.  these also dont seem to have a name in coala terminology.

We do have `DependencyBear`, which is the recipient of Result objects from a meta bear.